transport-native-io_uring/src/test/java/io/netty/channel/uring/IoUringSendZCTest.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.uring;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.MultiThreadIoEventLoopGroup;
+import io.netty.util.NetUtil;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class IoUringSendZCTest {
+
+    @Test
+    public void testSendZC() throws Exception {
+        String sampleString = "Hello Io_uring_send_zc";
+        MultiThreadIoEventLoopGroup group = new MultiThreadIoEventLoopGroup(1, IoUringIoHandler.newFactory());
+        try {
+            AtomicInteger targetResult = new AtomicInteger();
+            targetResult.set(sampleString.length());
+            LinkedBlockingQueue<ByteBuf> zcResult = new LinkedBlockingQueue<>();
+            ServerBootstrap serverBootstrap = new ServerBootstrap();
+            Channel serverChannel = serverBootstrap
+                    .channel(IoUringServerSocketChannel.class)
+                    .group(group)
+                    .childHandler(new ChannelInboundHandlerAdapter() {
+                        private CompositeByteBuf compositeByteBuf;
+
+                        @Override
+                        public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+                            compositeByteBuf = ctx.alloc().compositeBuffer();
+                        }
+
+                        @Override
+                        public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+                            if (compositeByteBuf != null) {
+                                compositeByteBuf.release();
+                            }
+                        }
+
+                        @Override
+                        public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+                            ByteBuf buf = (ByteBuf) msg;
+                            compositeByteBuf.addComponent(true, buf);
+                            if (compositeByteBuf.readableBytes() == targetResult.get()) {
+                                zcResult.put(compositeByteBuf);
+                                compositeByteBuf = ctx.alloc().compositeBuffer();
```

Comment:
just null it out ?

Reply:
These are for two different test cases. One meets the IO_URING_SEND_ZC_THRESHOLD, and the other does not meet the IO_URING_SEND_ZC_THRESHOLD.